### PR TITLE
feat: add starship as the prompt manager

### LINKED ISSUE #617: feature: Correct PROMPT prefix in the container terminal
## Description

```console
❯ envd up                                                                                                                            (base)
[+] ⌚ parse build.envd and download/cache dependencies 0.0s ✅ (finished)
 => 💽 (cached) download oh-my-zsh                                                                                                      0.0s
[+] build envd environment 203.8s (31/31) FINISHED
 => local://cache-dir                                                                                                                  0.5s
 => => transferring cache-dir: 426.41kB                                                                                                0.5s
 => docker-image://docker.io/tensorchord/python:3.8-ubuntu20.04                                                                        8.9s
 => => resolve docker.io/tensorchord/python:3.8-ubuntu20.04                                                                            8.9s
 => CACHED [internal] create user group envd                                                                                           0.0s
 => CACHED [internal] create user envd                                                                                                 0.0s
 => CACHED [internal] add user envd to sudoers                                                                                         0.0s
 => CACHED [internal] configure user permissions                                                                                       0.0s
 => CACHED [internal] configure user permissions                                                                                       0.0s
 => CACHED copy /oh-my-zsh /home/envd/.oh-my-zsh                                                                                       0.0s
 => CACHED mkfile /home/envd/install.sh                                                                                                0.0s
 => CACHED install oh-my-zsh                                                                                                           0.0s
 => CACHED mkfile /home/envd/.zshrc                                                                                                    0.0s
 => CACHED [internal] initialize conda bash environment                                                                                0.0s
 => CACHED [internal] create conda environment                                                                                         0.0s
 => CACHED [internal] initialize conda zsh environment                                                                                 0.0s
 => CACHED [internal] add conda environment to zshrc                                                                                   0.0s
 => CACHED [internal] create cache dir                                                                                                 0.0s
 => CACHED [internal] settings pip cache mount permissions                                                                             0.0s
 => CACHED mkdir /var/envd                                                                                                             0.0s
 => CACHED install ssh keys                                                                                                            0.0s
 => CACHED install ssh keys                                                                                                            0.0s
 => CACHED install system packages                                                                                                     0.0s
 => CACHED install conda packages                                                                                                      0.0s
 => pip install numpy                                                                                                                 18.5s
 => install PyPI packages                                                                                                              0.2s
 => merging all components into one                                                                                                  137.7s
 => => merging                                                                                                                       137.7s
 => update alternative python to envd                                                                                                  8.4s
 => update alternative python to envd                                                                                                  2.6s
 => update alternative python to envd                                                                                                  2.5s
 => update alternative python to envd                                                                                                  2.5s
 => mkfile /home/envd/.gitconfig                                                                                                       2.4s
 => exporting to oci image format                                                                                                     19.2s
 => => exporting layers                                                                                                               16.2s
 => => exporting manifest sha256:f604733ea4336a32be908470ad658e037b2183cae39ad66031e716435f494ba5                                      0.0s
 => => exporting config sha256:8136afd207792423d2682a75f3183ea937adbbb079412d97ff05253dae55c112                                        0.0s
 => => sending tarball                                                                                                                 3.0s
(envd) (envd) ➜  quick-start git:(master) ls
LICENSE  README.md  build.envd  demo.py
(envd) (envd) ➜  quick-start git:(master)
```

## Reproduction

run `envd up` with the `envd-quick-start`

## Additional Info

I guess one is `zsh` PROMPT, another is `conda` PROMPT.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

* close #617 
* related to #677 but starship will be better because users can get consistent experience under different shells

image

### CODE REVIEW

File: base-images/build.sh [hunk 1]
```diff
@@ -15,22 +15,22 @@ fi
...
     --build-arg ENVD_SSH_IMAGE=ghcr.io/tensorchord/envd-ssh-from-scratch \
     --build-arg HTTP_PROXY=${HTTP_PROXY} \
-     --build-arg HTTPS_PROXY=${HTTPS_PROXY} \
+    --build-arg HTTPS_PROXY=${HTTPS_PROXY} \
     --pull --push --platform linux/x86_64,linux/arm64 \
-    -t ${DOCKER_HUB_ORG}/python:3.8-ubuntu20.04-cuda11.6-cudnn8 \
-    -f python3.8-ubuntu20.04-cuda11.6.Dockerfile .
-docker buildx build \
+    -t ${DOCKER_HUB_ORG}/python:3.9-ubuntu20.04 \
+    -f python3.9-ubuntu20.04.Dockerfile .
```
Comment (terrytangyuan) [inline base-images/build.sh:25]: Consider parameterizing these to avoid redundant typing and reduce maintenance effort.

File: pkg/shell/zsh.go [hunk 2]
```diff
@@ -137,9 +137,9 @@ func (m generalManager) DownloadOrCache() (bool, error) {
 		return false, errors.Wrap(err, "failed to checkout master")
 	}
 
-	if err := m.createEnvdPromptTheme(); err != nil {
-		return false, errors.Wrap(err, "failed to create envd.zsh-theme")
-	}
+	// if err := m.createEnvdPromptTheme(); err != nil {
+	// 	return false, errors.Wrap(err, "failed to create envd.zsh-theme")
+	// }
```
Comment (terrytangyuan) [inline pkg/shell/zsh.go:142]: Remove commented code if no longer needed.

File: base-images/build.sh (hunk 1 above)
Comment (gaocegege) [reply base-images/build.sh:25]: I think we should create a new repo and automate the base image-building process.

**Review by gaocegege — APPROVED** [review]
Thanks for your contribution! :tada: :+1:

**Pull request merged**